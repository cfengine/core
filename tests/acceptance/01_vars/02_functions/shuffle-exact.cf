#######################################################
#
# Test shuffle()
#
#######################################################

body common control
{
      inputs => { "../../default.cf.sub" };
      bundlesequence  => { default("$(this.promise_filename)") };
      version => "1.0";
}

#######################################################

bundle agent init
{
  meta:
      "test_soft_fail" string => "!linux",
        meta => { "redmine7950" };

  vars:
      "a" slist => { "a", "b", "c", "d", "e", "f", "g" };
      "b" slist => { };
}

#######################################################

bundle agent test
{
  meta:
      # for some reason, shuffle() produces different results on 64bit RHEL 4
      # and Debian 4 than everywhere else
      "test_soft_fail"
        string => "(centos_4|debian_4).64_bit",
        meta => { "CFE-2301" };
  vars:
      "lists" slist => { "a", "b" };
      "seeds" slist => { "skruf", "cormorant", "dollhouse" };

      "shuffle_$(lists)_$(seeds)" slist => shuffle("init.$(lists)", $(seeds));
      "shuffle_inline_$(seeds)" slist => shuffle('["a", "b", "delta", "farmer"]', $(seeds));
}

#######################################################

bundle agent check
{
  methods:
      "check"  usebundle => dcs_check_state(test,
                                           "$(this.promise_filename).expected.json",
                                           $(this.promise_filename));
}
