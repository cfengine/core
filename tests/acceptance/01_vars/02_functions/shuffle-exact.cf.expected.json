{
  "lists": [
    "a",
    "b"
  ],
  "seeds": [
    "skruf",
    "cormorant",
    "dollhouse"
  ],
  "shuffle_a_cormorant": [
    "e",
    "d",
    "b",
    "a",
    "c",
    "g",
    "f"
  ],
  "shuffle_a_dollhouse": [
    "b",
    "f",
    "a",
    "e",
    "g",
    "c",
    "d"
  ],
  "shuffle_a_skruf": [
    "f",
    "d",
    "b",
    "e",
    "c",
    "g",
    "a"
  ],
  "shuffle_b_cormorant": [],
  "shuffle_b_dollhouse": [],
  "shuffle_b_skruf": [],
  "shuffle_inline_cormorant": [
    "delta",
    "a",
    "farmer",
    "b"
  ],
  "shuffle_inline_dollhouse": [
    "delta",
    "farmer",
    "a",
    "b"
  ],
  "shuffle_inline_skruf": [
    "b",
    "a",
    "delta",
    "farmer"
  ]
}
